client/memory.c

```diff
@@ -135,8 +139,8 @@ void *xdag_malloc(size_t size)
 {
 	uint8_t *res;
 	
-	if (size <= 0) {
-		return 0;
+	if (!size) {
```

Comment:
Can't size be less than zero?

Reply:
size_t is unsigned

Reply:
So what is the difference? If some guys change it to signed, what will happen? The size_t could be replaced by some guy which may be ignored by us. So I suggest to keep it as simple as possible to ensure the attack from hackers who may submit PRs which will cause security issue when combine them up.